job-scheduler/src/main/java/org/hawkular/metrics/scheduler/impl/JobsService.java

```diff
@@ -0,0 +1,149 @@
+import org.hawkular.metrics.scheduler.api.RepeatingTrigger;
+import org.hawkular.metrics.scheduler.api.SingleExecutionTrigger;
+import org.hawkular.metrics.scheduler.api.Trigger;
+import org.hawkular.rx.cassandra.driver.RxSession;
+import org.jboss.logging.Logger;
+
+import com.datastax.driver.core.KeyspaceMetadata;
+import com.datastax.driver.core.PreparedStatement;
+import com.datastax.driver.core.ResultSet;
+import com.datastax.driver.core.UDTValue;
+import com.datastax.driver.core.UserType;
+
+import rx.Observable;
+
+/**
+ * @author jsanda
+ */
+public class JobsService {
+
+    private static Logger logger = Logger.getLogger(JobsService.class);
+
+    private RxSession session;
+
+    private PreparedStatement findScheduled;
+
+    private PreparedStatement insertScheduled;
+
+    private PreparedStatement update;
+
+    private PreparedStatement updateStatus;
+
+    public JobsService(RxSession session) {
+        this.session = session;
+        findScheduled = session.getSession().prepare(
+                "SELECT job_id, job_type, job_name, job_params, trigger, status FROM scheduled_jobs_idx " +
+                "WHERE time_slice = ?");
+        update = session.getSession().prepare(
+                "INSERT INTO jobs (id, type, name, params, trigger) VALUES (?, ?, ?, ?, ?)");
+        insertScheduled =  session.getSession().prepare(
+                "INSERT INTO scheduled_jobs_idx (time_slice, job_id, job_type, job_name, job_params, trigger, " +
+                "status) VALUES (?, ?, ?, ?, ?, ?, ?)");
+        updateStatus = session.getSession().prepare(
+                "UPDATE scheduled_jobs_idx SET status = ? WHERE time_slice = ? AND job_id = ?");
+    }
+
+    public Observable<JobDetails> findScheduledJobs(Date timeSlice, rx.Scheduler scheduler) {
+        return session.executeAndFetch(findScheduled.bind(timeSlice), scheduler)
+                .map(row -> new JobDetails(
+                        row.getUUID(0),
+                        row.getString(1),
+                        row.getString(2),
+                        row.getMap(3, String.class, String.class),
+                        getTrigger(row.getUDTValue(4)),
+                        row.getByte(5)));
+    }
+
+    public Observable<ResultSet> insert(Date timeSlice, JobDetails job) {
+        return session.execute(insertScheduled.bind(timeSlice, job.getJobId(), job.getJobType(), job.getJobName(),
+                job.getParameters(), getTriggerValue(session, job.getTrigger())));
+    }
+
+    public Observable<ResultSet> updateStatusToFinished(Date timeSlice, UUID jobId) {
+        return session.execute(updateStatus.bind((byte) 1, timeSlice, jobId))
+                .doOnError(t -> logger.warn("There was an error updating the status to finished for [" + jobId +
+                        "] in time slice [" + timeSlice.getTime() + "]"));
+    }
+
+    Trigger getTrigger(UDTValue value) {
+        int type = value.getInt("type");
```

Comment:
Actually, it is described in the switch statement that follows:

```java
switch (type) {
            case 0:
                return new SingleExecutionTrigger(value.getLong("trigger_time"));
            case 1:
                return new RepeatingTrigger(
                        value.getLong("interval"),
                        value.getLong("delay"),
                        value.getLong("trigger_time"),
                        value.getInt("repeat_count"),
                        value.getInt("execution_count")
                );
            default:
                throw new IllegalArgumentException("Trigger type [" + type + "] is not supported");
        }
```

I am not sure that an enum adds much value here beyond documentation.